src/System.Management.Automation/engine/lang/scriptblock.cs

```diff
@@ -519,7 +519,10 @@ internal T InvokeAsMemberFunctionT<T>(object instance, object[] args)
                                        invocationInfo: null,
                                        propagateAllExceptionsToTop: true,
                                        args: args);
-            Diagnostics.Assert(result.Count == 1, "Code generation ensures we return the correct type");
+            if (result.Count == 0)
```

Comment:
I understand the change, but it feels really dirty because in an indirect way, it allows a method to not return a value, similar to the parser allowing the method to omit a `return` statement.

Unfortunately I can't think of a cleaner fix. I suppose it's worth adding a comment explaining that this is **not** to allow methods to omit a `return` statement.

Reply:
Yeah - it feels dirty to me but pipeline semantics make it necessary. I like the idea of adding a comment to explain it.